backend/src/schema/resolvers/searches.js

```diff
@@ -0,0 +1,69 @@
+const transformReturnType = record => {
+  return {
+    __typename: record.get('type'),
+    ...record.get('resource').properties,
+  }
+}
+export default {
+  Query: {
+    findResources: async (_parent, args, context, _resolveInfo) => {
+      const { query, limit } = args
+      const { id: thisUserId } = context.user
+      // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
+      const myQuery = query + '*'
+      const postCypher = `
+      CALL db.index.fulltext.queryNodes('post_fulltext_search', $query)
+      YIELD node as resource, score
+      MATCH (resource)<-[:WROTE]-(user:User)
+      WHERE score >= 0.5
+      AND NOT (user.deleted = true OR user.disabled = true
+      OR resource.deleted = true OR resource.disabled = true
+      OR (:User { id: $thisUserId })-[:BLOCKED]-(user))
+      RETURN resource, labels(resource)[0] AS type
+      LIMIT $limit
+      `
+      const session = context.driver.session()
+      let postResults, userResults
+      const readPostTxResultPromise = session.readTransaction(async transaction => {
+        const postTransactionResponse = transaction.run(postCypher, {
+          query: myQuery,
+          limit,
+          thisUserId,
+        })
+        return postTransactionResponse
+      })
+      try {
+        postResults = await readPostTxResultPromise
```

Comment:
We can improve performance here by waiting on both promises in parallel.

Reply:
I tried that and it threw an error saying that you could not start another transaction if there is already one pending...

Reply:
Implemented now.